src/main/scala/ch/openolitor/stammdaten/StammdatenInsertService.scala

```diff
@@ -738,60 +703,80 @@ class StammdatenInsertService(override val sysConfig: SystemConfig) extends Even
-              val sum = stammdatenWriteRepository.sumPreisTotalGeplanteLieferungenVorher(lieferung.vertriebId, lieferung.datum, project.get.geschaftsjahr.start(lieferung.datum.toLocalDate()).toDateTimeAtCurrentTime()).getOrElse(BigDecimal(0))
-
-              val durchschnittspreisBisher: BigDecimal = lieferungVorher.anzahlLieferungen match {
-                case 0 => BigDecimal(0)
-                case _ => sum / lieferungVorher.anzahlLieferungen
-              }
-              val anzahlLieferungenNeu = lieferungVorher.anzahlLieferungen + 1
-              (durchschnittspreisBisher, anzahlLieferungenNeu)
-            case _ =>
-              (BigDecimal(0), 1)
-          }
-          val lpId = Some(data.lieferplanungId)
-
-          val updatedLieferung = lieferung.copy(
-            lieferplanungId = lpId,
-            status = Offen,
-            durchschnittspreis = newDurchschnittspreis,
-            anzahlLieferungen = newAnzahlLieferungen,
-            modifidat = meta.timestamp,
-            modifikator = personId
-          )
+      stammdatenWriteRepository.getById(lieferplanungMapping, lieferungPlanungAdd.lieferplanungId) map { lieferplanung =>
+        stammdatenWriteRepository.getById(lieferungMapping, lieferungPlanungAdd.id) map { lieferung =>
+          updateLieferungUndZusatzLieferung(meta, lieferungPlanungAdd.lieferplanungId, project, lieferung)
+        }
+      }
+    }
+  }
 
-          //create koerbe
-          val adjustedLieferung = createKoerbe(updatedLieferung)
-
-          //update Lieferung
-          stammdatenWriteRepository.updateEntity[Lieferung, LieferungId](adjustedLieferung.id)(
-            lieferungMapping.column.lieferplanungId -> adjustedLieferung.lieferplanungId,
-            lieferungMapping.column.status -> adjustedLieferung.status,
-            lieferungMapping.column.durchschnittspreis -> adjustedLieferung.durchschnittspreis,
-            lieferungMapping.column.anzahlLieferungen -> adjustedLieferung.anzahlLieferungen,
-            lieferungMapping.column.anzahlKoerbeZuLiefern -> adjustedLieferung.anzahlKoerbeZuLiefern,
-            lieferungMapping.column.anzahlAbwesenheiten -> adjustedLieferung.anzahlAbwesenheiten,
-            lieferungMapping.column.anzahlSaldoZuTief -> adjustedLieferung.anzahlSaldoZuTief
-          )
+  private def updateLieferungUndZusatzLieferung(meta: EventMetadata, lieferplanungId: LieferplanungId, project: Option[Projekt], lieferung: Lieferung)(implicit personId: PersonId = meta.originator, session: DBSession, publisher: EventPublisher): Lieferung = {
+    val adjustedLieferung = offenLieferung(meta, lieferplanungId, project, lieferung)
+    stammdatenWriteRepository.getExistingZusatzAbotypen(adjustedLieferung.id).map { zusatzAbotyp =>
+      stammdatenWriteRepository.getExistingZusatzaboLieferung(zusatzAbotyp.id, lieferplanungId) match {
```

Comment:
We must check here not only on the 'lieferplanungId' but also on the 'datum' because we must create multiple Lieferungen in case the same Zusatzabotyp exists within Abos in Lieferungen with different dates but in the same Lieferplanung.

Reply:
Done